components/Article/Extract.js

```diff
@@ -0,0 +1,46 @@
+import React, { Fragment } from 'react'
+import Head from 'next/head'
+import { renderMdast } from 'mdast-react-render'
+
+const parseRanges = ranges => (
+  ranges.split(':').map(range => {
+    let [start, end] = range.split('...')
+    start = +start || 0
+
+    if (end === '') {
+      end = undefined
```

Comment:
This logic might is a bit more obvious if it's clear the values end up in Array.slice(), so parseSlices() would sound clearer to me.

Reply:
improved naming and added tests